inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,348 @@
+            return
+        if position == len(self.keypath) - 1:
+            if isinstance(record[key], list):
+                if self.match_type == 'regex':
+                    record[key] = [self.value if re.search(
+                        self.value_to_check, x) else x for x in record[key]]
+
+                elif self.match_type == 'exact':
+                    record[key] = [self.value if x == self.value_to_check else x for x in record[key]]
+
+                elif self.match_type == 'contains':
+                    record[key] = [self.value if self.value_to_check in x else x for x in record[key]]
+                self.changed = True
+            else:
+                if self.match_type == 'exact' and record[key] == self.value_to_check:
+                    record[key] = self.value
+                if self.match_type == 'regex' and re.search(
+                        self.value_to_check, record[key]):
+                            record[key] = self.value
+                elif self.match_type == 'contains' and self.value_to_check in record[key]:
+                    record[key] = self.value
+                self.changed = True
+                return
+        else:
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply(array_record, schema, position + 1, conditions_passed)
+            else:
+                self.apply(record[key], schema, position + 1, conditions_passed)
+
+
+def create_schema_record(schema, path, value):
+    """Object creation in par with the schema."""
+    record = {}
+    temp_record = record
+    new_schema = schema
+    if new_schema['type'] == 'array':
+        new_schema = new_schema['items']['properties']
+    elif new_schema['type'] == 'object':
+        new_schema = new_schema['properties']
+    for key in path:
+        new_schema = new_schema[key]
+        if new_schema['type'] == 'object':
+            new_schema = new_schema['properties']
+            temp_record[key] = {}
+            temp_record = temp_record[key]
+
+        elif new_schema['type'] == 'array':
+            if new_schema['items']['type'] == 'object':
+                new_schema = new_schema['items']['properties']
+            if len(path) == 1:
+                temp_record[key] = [value]
+                return record
+            else:
+                temp_record[key] = [{}]
+                temp_record = temp_record[key][0]
+    temp_record[path[-1]] = value
+    return record
+
+
+def check_value(record, match_type, keypath, value_to_check, position):
+    """Function that checks the validity of the condition."""
+    key = keypath[position]
+    if not record.get(key):
+        return match_type == 'missing'
+    temp_record = record[key]
+    if isinstance(temp_record, list):
+        for index, array_record in enumerate(temp_record):
+            if position + 1 == len(keypath):
+                if match_type == 'exact' and array_record == value_to_check:
```

Comment:
would be better to have a map where key is `match_type` and value is `match_function` so that we don't have to use `if` and more importantly duplicate them.

Example:
```python
{
  'exact': lambda a, b: a == b
...
}
```